packages/@aws-cdk/aws-iam/lib/immutable-role.ts

```diff
@@ -0,0 +1,43 @@
+import { Grant } from './grant';
+import { IManagedPolicy } from './managed-policy';
+import { Policy, PolicyProps } from './policy';
+import { PolicyStatement } from './policy-statement';
+import { IPrincipal } from './principals';
+import { IRole } from './role';
+
+export class ImmutableRole implements IRole {
```

Comment:
Documentation is missing - what this is and why we need it

Reply:
Done.